tests/Aspire.Hosting.Elasticsearch.Tests/ElasticsearchPublicApiTests.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Xunit;
+
+namespace Aspire.Hosting.Elasticsearch.Tests;
+
+public class ElasticsearchPublicApiTests
+{
+    [Fact]
+    public void AddElasticsearchContainerShouldThrowWhenBuilderIsNull()
+    {
+        IDistributedApplicationBuilder builder = null!;
+        const string name = "Elasticsearch";
+
+        var action = () => builder.AddElasticsearch(name);
+
+        var exception = Assert.Throws<ArgumentNullException>(action);
+        Assert.Equal(nameof(builder), exception.ParamName);
+    }
+
+    [Fact]
+    public void AddElasticsearchContainerShouldThrowWhenNameIsNull()
+    {
+        IDistributedApplicationBuilder builder = new DistributedApplicationBuilder([]);
```

Comment:
As I found out. Better to use TestDistributedApplicationBuilder.Create();

Reply:
AFIK `TestDistributedApplicationBuilder` is suitable when starting the app. Here, it seems unnecessary.

Reply:
> AFIK `TestDistributedApplicationBuilder` is suitable when starting the app. Here, it seems unnecessary.

I wanted this more from the point of view that there would be a single code base for checking the “public zero”.
Although you may be right. View this at your own discretion. Well, or ask the guys what they think about this